### DISCUSSION TITLE
The results read by Panoply and ToolSUI-5.5.2 are quite different

### DISCUSSION THREAD

**Discussion by kmood** [Unidata/netcdf-java#1081] [Questions and Answers] [answered]
**netCDF information:**
```
  title= null
    id= null
    fileType= NetCDF-4
    fileDesc= Hierarchical Data Format, version 5
    fileVersion= N/A
    class= ucar.nc2.NetcdfFile
    iosp= class ucar.nc2.internal.iosp.hdf5.H5iospNew
   raf = /OHF.2020051513.nc
   size= 3306837 (3.306 Mb)
  
  Variable isCaching  size     cachedSize (bytes) 
   lon       true      564 
   lat       true      484 
   time      true      192 
   T        false   818928 
   Pr       false   818928 
   Rh       false   818928 
   Cloud    false   818928 
   Vis      false   818928 
   UM       false   818928 
   VM       false   818928 
   WW       false   818928 
                             --------
            total        6 Mb cached=        0 Kb
```
**NetCDF file variable information :**
```
 short Pr(time=24, lat=121, lon=141);
      :least_significant_digit = 3L; // long
      :description = "Pr";
      :add_offset = 500.0; // double
      :scale_factor = 0.030525; // double
      :missing_value = 32767S; // short
      :units = "mm";
      :_ChunkSizes = 1U, 121U, 141U; // uint
```
**Panoply and ToolSUI-5.5.2 Comparison between the Read Result :**
![image](https://user-images.githubusercontent.com/31915216/190080734-fd8324dc-f159-4ff2-ba09-b465128b68e5.png)
![image](https://user-images.githubusercontent.com/31915216/190081492-1323feee-2ed5-43ce-9a7d-be9e5f435dac.png)

I failed to find the possible reason through consulting the materials, and hope to give some help ！！

**Comment by rschmunk** [comment, marked as answer]
Panoply is using the `scale_factor` and `add_offset` attributes to unpack the data before it shows you the values. It appears to me that toolsUI is showing you the packed data.

Note that -16380 * 0.030525 + 500 = 0.0005, which if shown with %0.1f formatting is 0.0.

**Reply by haileyajohnson** [reply]
I suspect this is the same as [TDS issue 282](https://github.com/Unidata/tds/issues/282)

**Reply by kmood** [reply]
I did not look at the source code. I suspect that the variable type is short, but the variable contains scale_factor and add_offset attributes, which leads to the inconsistency of the final result. I am not sure whether this is a bug @haileyajohnson

**Reply by rschmunk** [reply]
In that second screen cap of yours, it says the variable type is short.

**Reply by kmood** [reply]
Is this a bug? Whether it can be repaired @rschmunk

**Reply by rschmunk** [reply]
I think the question is, what values is ToolsUI supposed to display? The enhanced/unpacked values? Or the unenhanced/packed/original values?

Panoply displays the values that it plots. I don't know what the goal is for ToolsUI. Only someone at Unidata can answer that.

**Reply by ethanrd** [reply]
I believe the default for the "Viewer" window in ToolsUI is to show the data as it is in the dataset. You can change that with the decorative "flowers in a vase" icon/button next to the folder icon. It turns on the various enhancements. You have to reload the dataset to see the changes.

Not an intuitive icon/button and the mouse-over hint, "add Coordinates is [OFF|ON]", isn't much better.

**Comment by ethanrd** [comment]
I believe the default for the "Viewer" window in ToolsUI is to show the data as it is in the dataset. You can change that with the decorative "flowers in a vase" icon/button next to the folder icon. It turns on the various enhancements. You have to reload the dataset to see the changes.

Not an intuitive icon/button and the mouse-over hint, "add Coordinates is [OFF|ON]", isn't much better.

**Reply by ethanrd** [reply]
Oops! Meant for this to be a reply in @rschmunk 's answer above.
